### PR TITLE
feat(mcdu): add radnav template page

### PR DESCRIPTION
**Pull request by samiFBG**
Fixes #[issue_no]

## Summary of Changes

      I have add the template of rnav page in the new mcdu
## Screenshots (if necessary)
alt text

## References

I have use Honeywell Pegasus Step 1A (Rev 0), 2009 edition manual. 
alt text

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/instruments/src/MCDU/Pages/PagesContainer.tsx [hunk 1]
```diff
@@ -27,6 +27,8 @@ import IdentPage from './FMGC/Ident/Ident';
 import { PAGE_KEYS } from '../Components/Buttons';
 import FuelPred from './FMGC/Fuel/FuelPred';
 import FplanHolder from './FMGC/Fplan/Fplan';
+import RadnavPage from './FMGC/RadNav/Radnav';
+import { addScratchpadMessage } from '../redux/actions/scratchpadActionCreators';
```
Comment (Lucky38i) [inline src/instruments/src/MCDU/Pages/PagesContainer.tsx:31]: This is the incorrect way to be using redux hooks. this action creator by itself won't do anything you need to dispatch it. Also, this is the pages container but I wouldn't use it as a redux container, better you do that inside of `Radnav.tsx`

File: src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+            color={lineColors.inop}
+        />
+    </LineHolder>
+);
+type RadnavePageProps = {
+    addMessage: (msg: ScratchpadMessage) => void,
+}
+
+const RadnavPage: React.FC<RadnavePageProps> = ({ addMessage }) => {
+    const dispatch = useMCDUDispatch();
```
Comment (Lucky38i) [inline src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx:346]: in reference to my other comment, you already have the dispatcher here so just do 
```js
import * as scratchpadActions from '../../../redux/actions/scratchpadActionCreators';
const addMessage = (msg: ScratchpadMessage) => {
        dispatch(scratchpadActions.addScratchpadMessage(msg));
};
```
See below for a usage example https://github.com/flybywiresim/a32nx/blob/1bdbc95860f8b482c0928e3e349256d74aa6ef84/src/instruments/src/MCDU/Pages/FMGC/Init/Init.tsx#L56-L58

File: src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        dispatch(titlebarActions.setTitleBarText(msg));
+    });
+    useEffect(() => {
+        setTitlebar(' RADIO NAV ');
+    }, []);
+    return (
+        <>
+            <RowHolder columns={2}>
+                <Vorone addMessage={addMessage} />
+                <Vortwo addMessage={addMessage} />
```
Comment (Lucky38i) [inline src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx:357]: Consider merging `Vorone` and `vortwo` into one single react component? makes the code tidier and the uses for both cases are the same. The same for `crsone` and `crstwo`, `ls` and `lscrs`, as-well as `adfone` and `adftwo`.

Bear in mind a single component can return `RowHolder`s so with the case of merging `ls` and `lscrs` you can do
```js
return (
    <RowHolder columns={2}>
        <...the inners of ls component/>
     </RowHolder>
     <RowHolder columns={2}>
         <...the innerds of lscrs component />
     </RowHolder>
)
```

**Review by Lucky38i — CHANGES_REQUESTED** [review]
Very nice template so far, just a few changes

**Comment by samiFBG** [comment]
> Fixes #[issue_no]
> ## Summary of Changes
> 
> ```
>   I have add the template of rnav page in the new mcdu
> ```
> 
> ## Screenshots (if necessary)
> 
> alt text
> ## References
> 
> I have use Honeywell Pegasus Step 1A (Rev 0), 2009 edition manual. alt text
> ## How to download the PR for QA
> 
> Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.
> 
>     1. Make sure you are signed in to GitHub
> 
>     2. Click on the **Checks** tab on the PR
> 
>     3. On the left side, click on the bottom **PR** tab
> 
>     4. Click on the **A32NX** download link at the bottom of the page

File: src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    const setNewValue = (value: string | undefined) => {
+        if (value === undefined) {
+            setLscrs(undefined);
+        } else {
+            setLscrs(value);
+        }
+    };
+
+    return (
+        <LineHolder columnPosition={1}>
```
Comment (Lucky38i) [inline src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx:179]: I think you forgot to add a RowHolder here

**Review by Lucky38i — CHANGES_REQUESTED** [review]
Much better changes, just double check the rendering in-sim. looks like you might've forgot a LineHolder or RowHolder

File: src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+    useEffect(() => {
+        setTitlebar(' RADIO NAV ');
+    }, []);
+    return (
+        <>
+            <Vor addMessage={addMessage} clearMessage={clearMessage} />
+            <Crs clearMessage={clearMessage} />
+            <Ls addMessage={addMessage} clearMessage={clearMessage} />
+            <Lscrs clearMessage={clearMessage} />
+            <Adf addMessage={addMessage} clearScratchpad={clearMessage} />
```
Comment (Lucky38i) [inline src/instruments/src/MCDU/Pages/FMGC/RadNav/Radnav.tsx:328]: All interactive components automatically clear the scratchpad, you don't need to handle that. https://github.com/flybywiresim/a32nx/blob/1bdbc95860f8b482c0928e3e349256d74aa6ef84/src/instruments/src/MCDU/Components/Fields/Interactive/StringInputField.tsx#L55-L62

**Review by Lucky38i — CHANGES_REQUESTED** [review]
gotta remove the clearing of the scratchpad in the page, the components do this automatically.

As a page builder, you need to handle invalid values, setting values and when the user presses `CLR` on the scratchpad. Below is an example, 
https://github.com/flybywiresim/a32nx/blob/1bdbc95860f8b482c0928e3e349256d74aa6ef84/src/instruments/src/MCDU/Pages/FMGC/Init/InitA.tsx#L166-L184

**Comment by St54Kevin** [comment]
VHF and ADF empty fields should be one space wider. The devider is always at the same location as in the header

**Comment by davidwalschots** [comment]
@samiFBG Do you intend to continue work on this PR in the near future?

**Comment by davidwalschots** [comment]
I'm going to close this PR, because the author is not responding. Feel free to (re)open.

**Pull request closed without merging**